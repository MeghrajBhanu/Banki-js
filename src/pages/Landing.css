.bank-card{
    height:200px;
    width:300px;
    margin:10px;
}
.wrapper-card{
        display: flex;
        flex-wrap:wrap;
}