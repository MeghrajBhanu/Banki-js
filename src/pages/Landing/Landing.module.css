
  
h1 {
    font-family: 'Montserrat', sans-serif;
    font-size: 3em;
    margin-bottom: 0.2em;
    line-height: 1.2;
    color: #222
  }
  
.h3-h {
    font-weight: 400;
    color: #555;
    width: 30em;
  }
  
  .cards {
    display: flex;
    align-items: flex-start;
    flex-wrap: wrap;
    justify-content: space-around;
  }
  
  .card {
    background: #fff;
    width: 20em;
    border-radius: 0.8em;
    margin: 1em;
    overflow: hidden;
    cursor: pointer;
    box-shadow: 0 13px 27px -5px hsla(240, 30.1%, 28%, 0.25), 0 8px 16px -8px hsla(0, 0%, 0%, 0.3), 0 -6px 16px -6px hsla(0, 0%, 0%, 0.03);
    transition: all ease 200ms;
  }
  
  .card:hover {
    transform: scale(1.03);
    box-shadow: 0 13px 40px -5px hsla(240, 30.1%, 28%, 0.12), 0 8px 32px -8px hsla(0, 0%, 0%, 0.14), 0 -6px 32px -6px hsla(0, 0%, 0%, 0.02);
  }
  
  .card img {
    width: 100%;
    object-fit: cover;
  }
  
  .card h2 {
    color: #222;
    margin-top: -0.2em;
    line-height: 1.4;
    font-size: 1.3em;
    font-weight: 500;
    font-family: 'Montserrat', sans-serif;
    transition: all ease-in 100ms;
  }
  
  .card p {
    color: #777;
  }
  
  .card h5 {
    color: #bbb;
    font-weight: 700;
    font-size: 0.7em;
    letter-spacing: 0.04em;
    margin: 1.4em 0 0 0;
    text-transform: uppercase;
  }
  
  .card-body {
    padding: 1.2em;
    align-items: center;
  }
.pagnation{
    padding:10px;
    margin: 15px 0;
    display:flex;
    justify-content: center;
    align-items: center;
  }
.pagination > span{
    padding:15px 20px;
    border: 1px solid wheat;
    cursor:pointer;
}
.selected{
    background-color: rgb(206, 168, 121);
}
.disable_button{
  visibility:hidden;
}