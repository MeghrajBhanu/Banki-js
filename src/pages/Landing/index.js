import React, { useState, useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import {
  get_all_pancard,
  get_all_sort_fixed_depo,
  get_all_sort_account_type,
} from "../../redux/actions/pan_actions";
import BankAccountsMap from "../../components/BankAccountsMap";
import "bootstrap/dist/css/bootstrap.min.css";
import DropdownButton from "react-bootstrap/DropdownButton";
import Dropdown from "react-bootstrap/Dropdown";
import CardsShimmer from "../../components/CardsShimmer/CardsShimmer";
import classes from "./index.module.css";
import SharedHeader from "../../components/SharedHeader";

/**
 * Landing component displays users homepage after loggingin
 *
 * @returns {React.ReactElement}
 */

const Landing = () => {
  const user = useSelector((state) => state.auth.user);

  const dispatch = useDispatch();
  const data = useSelector((state) => state.pan.data);
  const [selected, setSelected] = useState({});
  const isLoading = useSelector((state) => state.pan.isLoading);
  const error = useSelector((state) => state.pan.isLoading);
  const [page, setPage] = useState(1);

  const handleSelect = (key, event) => {
    setSelected({ key, value: event.target.value });
  };
  useEffect(() => {
    dispatch(get_all_pancard(user?.pancard ? user["pancard"] : "LPIOM1234L"))
      .then((data) => {
        console.log("success");
      })
      .catch((err) => {
        alert(err);
      });
  }, []);
  useEffect(() => {
    if (selected === {} || selected.key === "None")
      dispatch(get_all_pancard(user?.pancard ? user["pancard"] : "LPIOM1234L"));

    if (selected.key === "FixedDeposits") {
      dispatch(get_all_sort_fixed_depo(user?.pancard ? user["pancard"] : "LPIOM1234L"));
      setTimeout(()=>{
          console.log(data)
      },600)
    }
    if (selected.key === "Account_Type: Salary") {
      dispatch(get_all_sort_account_type(user["pancard"], "Salary"));
    }
    if (selected.key === "Account_Type: Savings") {
      dispatch(get_all_sort_account_type(user["pancard"], "Savings"));
    }
    setPage(1);
  }, [selected]);

  const list = [
    { key: "Filter By" },
    { key: "None", value: "none" },
    { key: "FixedDeposits", value: "fixed_deposits" },
    { key: "Account_Type: Salary", value: "account_type:salary" },
    { key: "Account_Type: Savings", value: "account_type:savings" },
  ];

  let mapped = data?.slice(page * 10 - 10, page * 10).map((item) => {
    return <BankAccountsMap
        key={item._id}
        id={item._id}
        name={item.name}
        email={item.email}
        bankName={item.bankName}
        pancard={item.pancard}
        account_type={item.AccountType}
        fd={item.FixedDeposits}
        balance={item.Balance}
      />
    
  });
  const selectPageHandler = (page_i) => {
    if (
      page_i >= 1 &&
      page_i <= [...Array(Math.ceil(data.length / 10))].length &&
      page_i !== page
    )
      setPage(page_i);
  };
  return (
    <React.Fragment key="overallwrapper">
      {isLoading ? (
        <CardsShimmer key="cardshimmer"></CardsShimmer>
      ) : (
        <React.Fragment key="renderer">
          {error ? (
            <div key="errorrenderer">error</div>
          ) : (
            <React.Fragment key="mainrenderer">
              <SharedHeader name={user?.name} pancard={user?.pancard} key="ShareHeader"/>
              <div className="mb-4 a-section a-spacing-small mt-1 bg-light">
                <div className={classes["flex"]} key="flexwrapper">
                  <span key="pageresults">
                    {(page - 1) * 10 + 1}-
                    {Math.min((page - 1) * 10 + 10, data.length)} of over{" "}
                    {data.length} results for <span> </span>
                  </span>

                  <span key="pancardwarnig" className="text-warning" style={{ flex: "2" }}>
                    {" "}
                    "{user?.pancard}"
                  </span>
                  <span key="spandrop" className={classes["absolute"]}>
                    <DropdownButton
                      id="dropdown-basic-button"
                      key="dropdown-basic-button"
                      variant="secondary"
                      className="float-right"
                      style={{ marginRight: "25px" }}
                      onSelect={handleSelect}
                      data-testid="select-option"
                      size="sm"
                      title={selected?.key || list[0].key}
                    >
                      {list.map((item, index) => {
                        return (
                          <Dropdown.Item key={index} eventKey={item.key}>
                            {item.value}
                          </Dropdown.Item>
                        );
                      })}
                    </DropdownButton>
                  </span>
                </div>
              </div>
              {data?.length > 0 ? (
                <React.Fragment key="pagination+cardss">
                  <div className={classes["cards"]} key="cardss" >{mapped}</div>
                  <div
                    className={classes["pagination"]}
                    key="pgination"
                    style={{
                      padding: "10px",
                      margin: "15px 0",
                      display: "flex",
                      justifyContent: "center",
                      alignItems: "center",
                    }}
                  >
                    <span
                      onClick={() => selectPageHandler(page - 1)}
                      key="leftarrow"
                      className={page > 1 ? "" : classes["disable_button"]}
                    >
                      ◀
                    </span>

                    {[...Array(Math.ceil(data.length / 10))].map((_, i) => {
                      return (
                        <span
                          key={i}
                          className={page === i + 1 ? classes["selected"] : ""}
                          onClick={() => selectPageHandler(i + 1)}
                        >
                          {i + 1}
                        </span>
                      );
                    })}

                    <span
                    key="rightarrow"
                      onClick={() => selectPageHandler(page + 1)}
                      className={
                        page < data.length / 10 ? "" : classes["disable_button"]
                      }
                    >
                      ▶
                    </span>
                  </div>
                </React.Fragment>
              ) : (
                <h5 className="text-center m-3 text-secondary" key="noaccountsfound">
                  Couldnt find any accounts .
                </h5>
              )}
            </React.Fragment>
          )}
        </React.Fragment>
      )}
    </React.Fragment>
  );
};

export default Landing;
