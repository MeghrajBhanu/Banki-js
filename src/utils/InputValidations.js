export const validateEmail = (checkingText) => {
    
    const regexp = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/; 
   
    
    if (regexp.exec(checkingText) !== null) {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Email Address'
            };
        }
}
export const validatePan = (checkingText) => {
    
    
    if (checkingText.length === 10) {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Pan Number'
            };
        }
}
export const validatePass = (checkingText) => {
    
    
    if (checkingText!=="") {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Password'
            };
        }
}