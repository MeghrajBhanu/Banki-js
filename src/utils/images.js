import bank1 from "../assets/bank1.jpg";
import bank2 from "../assets/bank2.png";
import bank3 from "../assets/bank3.jfif";
import bank4 from "../assets/bank4.png";

const image_map={
    "Bank1":bank1,
    "Bank2":bank2,
    "Bank3":bank3,
    "Bank4":bank4,
}
export default image_map;