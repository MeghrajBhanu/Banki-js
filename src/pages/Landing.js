import React, { useState, useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { get_all_pancard,get_all_sort_fixed_depo,get_all_sort_account_type } from "../redux/actions/pan_actions";
import BankAccountsMap from "../components/BankAccountsMap";
import "bootstrap/dist/css/bootstrap.min.css";
import DropdownButton from "react-bootstrap/DropdownButton";
import Dropdown from "react-bootstrap/Dropdown";
import CardsShimmer from "../components/CardsShimmer";
import "./Landing.css";
const Landing = () => {
  const user = useSelector((state) => state.auth.user);

  const dispatch = useDispatch();
  const data = useSelector((state) => state.pan.data);
  console.log(data)
  const [selected, setSelected] = useState({});
  const isLoading=useSelector((state) => state.pan.isLoading);
  const handleSelect = (key, event) => {
    setSelected({ key, value: event.target.value });
  };
  useEffect(() => {
    
  
    dispatch(get_all_pancard(user["pancard"]))
    
  },[]
  )
  useEffect(() => {
    
    
    if(selected==={} || selected.key==="None")dispatch(get_all_pancard(user["pancard"]));
    
    if(selected.key==="FixedDeposits"){
      dispatch(get_all_sort_fixed_depo(user["pancard"]))
      
    }
    if(selected.key==="Account_Type: Salary"){

      dispatch(get_all_sort_account_type(user["pancard"],"Salary"))
    }
    if(selected.key==="Account_Type: Savings"){
      dispatch(get_all_sort_account_type(user["pancard"],"Savings"))
    }
    
  }, [selected]);
  
  
  
  const list = [
    {key:"Filter By"},
    { key: "None", value: "none" },
    { key: "FixedDeposits", value: "fixed_deposits" },
    { key: "Account_Type: Salary", value: "account_type:salary" },
    { key: "Account_Type: Savings", value: "account_type:savings"}
  ];

  let mapped = data?.map((item) => {
    return (
      <BankAccountsMap
        key={item._id}
        id={item._id}
        name={item.name}
        email={item.email}
        bankName={item.bankName}
        pancard={item.pancard}
        account_type={item.AccountType}
        fd={item.FixedDeposits}
        balance={item.Balance}
      ></BankAccountsMap>
    );
  });
  
  
    return(
    <React.Fragment>
      
      {isLoading?<CardsShimmer></CardsShimmer>:
      (
        <>
      <div class="a-section a-spacing-small mt-1 bg-light">
        <span>
          1-{data.length} of over {data.length} results for
        </span>
        <span> </span>
        <span className="text-warning">"{user.pancard}"</span>
        <span>
          <DropdownButton
            id="dropdown-basic-button"
            variant="secondary"
            className="float-right"
            onSelect={handleSelect}
            size="sm"
            title={selected?.key || list[0].key}
          >
            {list.map((item, index) => {
              return (
                <Dropdown.Item key={index} eventKey={item.key}>
                  {item.value}
                </Dropdown.Item>
              );
            })}
          </DropdownButton>
        </span>
      </div>

      <h1 className="text-center m-1 text-secondary ">Welcome {user.name}</h1>
      <h6 className="text-center m-3 text-secondary">Here are your bank accounts linked with {user.pancard} </h6>
      {data?.length>0 ?<div class="wrapper-card">{mapped}</div>:<h5 className="text-center m-3 text-secondary">Couldnt find any accounts .</h5>}
</>)}
          
    </React.Fragment>
    );
          
          
};
export default Landing;
