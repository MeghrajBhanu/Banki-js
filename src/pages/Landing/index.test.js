import { screen, waitFor } from "@testing-library/react";
import { renderComponent, onlywrap } from "../../utils/test_helper";
import Landing from ".";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";
const user = userEvent.setup();
describe("Landing Page", () => {
  test("snapshot", () => {
    const tree = renderer.create(onlywrap(<Landing />)).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("should render the  shared header anf filter button", () => {
    renderComponent(<Landing />);
    const welcome = screen.findByText(/welcome/i);
    const filter = screen.findByRole("button", {
      name: /filter by/i,
    });
    waitFor(() => {
      expect(welcome).toBeInTheDocument();
      expect(filter).toBeInTheDocument();
    });
  });
  test("data in form ofcards should be rendered", () => {
    renderComponent(<Landing />);

    const AccountType = screen.findByText(/salary/i);
    const bankName = screen.findByText(/Bank3/i);
    const view = screen.findByRole("button", {
      name: /view more/i,
    });
    waitFor(() => {
      expect(AccountType).toBeInTheDocument();
      expect(bankName).toBeInTheDocument();
      expect(view).toBeInTheDocument();
    });
  });
  test("redux store should recieve data", () => {
    let { store } = renderComponent(<Landing />);
    expect(store.getState().pan.data.length >= 1);
  });
  test("pagination", () => {
    renderComponent(<Landing />);
    waitFor(() => {
      expect(screen.getByText(/â—€/i)).toBeInTheDocument();
    });
  });
  test("Should handler filterrequests", () => {
    const { store } = renderComponent(<Landing />);
    const filter = screen.findByRole("button", {
      name: /filter by/i,
    });
    const fixed_deposits = screen.findByText("fixed_deposits");

    waitFor(() => {
      user.change(filter, { target: { value: "fixed_deposits" } });
      expect(fixed_deposits).toBeInTheDocument();
    
      expect(store.getState().pan.data.length ).toBe(0);
    });
  });
  test("check navbar",()=>{
    renderComponent(<Landing />)
    const landing=screen.findByText(/Landing/i)
    const flagged=screen.findByText(/Flagged/i)
    waitFor(()=>{
      expect(landing).toBeInTheDocument()
      expect(flagged).toBeInTheDocument()
    })
    
  })
});
