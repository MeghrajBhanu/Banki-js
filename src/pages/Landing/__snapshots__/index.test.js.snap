// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page snapshot 1`] = `
Array [
  <div
    className="title"
  />,
  <div
    className="sort_bar"
  />,
  <div
    className="cards"
  >
    <div
      className="card"
    >
      loading..
    </div>
    <div
      className="card"
    >
      loading..
    </div>
    <div
      className="card"
    >
      loading..
    </div>
    <div
      className="card"
    >
      loading..
    </div>
    <div
      className="card"
    >
      loading..
    </div>
    <div
      className="card"
    >
      loading..
    </div>
    <div
      className="card"
    >
      loading..
    </div>
    <div
      className="card"
    >
      loading..
    </div>
    <div
      className="card"
    >
      loading..
    </div>
    <div
      className="card"
    >
      loading..
    </div>
  </div>,
]
`;
